### DISCUSSION TITLE
Connect to sqlserver2000,throw errror: does not support encryption

### DISCUSSION THREAD

**Discussion by mingrenliu** [dotnet/SqlClient#2496] [Q&A] [answered]
**Question:** when i use  sqlconnection to connect sqlserver2000，throw error : The instance of SQL Server you attempted to connect to does not support encryption. connectionString: Data Source=**;User ID=** ;Password=** ;Initial Catalog=ScaleDb;TrustServerCertificate=true;even i change TrustServerCertificate to false or add encrypt=false (throw error: Internal connection fatal error ),both not work 
**version:** Microsoft.Data.SqlClient, Version=4.0.0.0; sqlserver: Microsoft SQL Server  2000 - 8.00.2039 (Intel X86)

**Comment by JRahnama** [comment, marked as answer]
Just a heads-up, anything before SQL Server 2016 is no longer supported, so it's a good idea to consider upgrading to a newer version for the best security and support.

Now, onto your question! Are you using TLS 1.3 or TLS 1.2 on your client machine? It's worth noting that SQL Server 2000 doesn't support TLS 1.2, while TLS 1.2 is available for SQL Server 2008 and later versions. Also, TLS 1.3 kicks in from SQL Server 2022 onwards, each with their own cipher suites.

If you're looking for a workaround, you could enable TLS 1.1 on your client machine, though it's not the safest option since it's deprecated. However, if you need to, you can still enforce its usage in your application.

**Reply by mingrenliu** [reply]
thank you . but  it's caused by TLS ? I have countered a problem caused by tls,but it throw different error ( sorry i do not remenber the  error detail ). Priviously, i use following solution to resolve: deploy a microservice in  inner network computer,use this service  to connect sqlserver (enable tls1.0) and outside applicament  request this service, because the database only for limited work.
